packages/principal/src/bls/type.ts

```diff
@@ -0,0 +1,76 @@
+import {
+  SignerKey,
+  VerifierKey,
+  MulticodecCode,
+  ByteView,
+  DIDKey,
+  SignatureView,
+  Await,
+} from '@ucanto/interface'
+import * as Signature from '@ipld/dag-ucan/signature'
+
+export * from '@ucanto/interface'
+
+/**
+ * Integer corresponding to BLS12381G2 byteprefix of the VarSig.
+ */
+export type SigAlg = MulticodecCode<typeof Signature.BLS12381G2, 'BLS12381G2'>
+
+/**
+ * Name corresponding to BLS12381G2 algorithm.
+ */
+export type Name = 'BLS12381G2'
+
+/**
+ * This interface parametrizes {@link SignerKey} and extends it with Ed specific
```

Comment:
with Ed specific looks copy paste and not needed?

Reply:
```suggestion
 * This interface parametrizes {@link SignerKey} and extends it with BLS specific
```